### PR TITLE
IGNITE-23211 Fix AssertionError on IgniteCache#get

### PR DESCRIPTION
**Pull request by nizhikov**
… EntryProcessor in optimistic transaction

Thank you for submitting the pull request to the Apache Ignite.

In order to streamline the review of the contribution 
we ask you to ensure the following steps have been taken:

### The Contribution Checklist
- [ ] There is a single JIRA ticket related to the pull request. 
- [ ] The web-link to the pull request is attached to the JIRA ticket.
- [ ] The JIRA ticket has the _Patch Available_ state.
- [ ] The pull request body describes changes that have been made. 
The description explains _WHAT_ and _WHY_ was made instead of _HOW_.
- [ ] The pull request title is treated as the final commit message. 
The following pattern must be used: `IGNITE-XXXX Change summary` where `XXXX` - number of JIRA issue.
- [ ] A reviewer has been mentioned through the JIRA comments 
(see [the Maintainers list](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute#HowtoContribute-ReviewProcessandMaintainers)) 
- [ ] The pull request has been checked by the Teamcity Bot and 
the `green visa` attached to the JIRA ticket (see [TC.Bot: Check PR](https://mtcga.gridgain.com/prs.html))

### Notes
- [How to Contribute](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute)
- [Coding abbreviation rules](https://cwiki.apache.org/confluence/display/IGNITE/Abbreviation+Rules)
- [Coding Guidelines](https://cwiki.apache.org/confluence/display/IGNITE/Coding+Guidelines)
- [Apache Ignite Teamcity Bot](https://cwiki.apache.org/confluence/display/IGNITE/Apache+Ignite+Teamcity+Bot)

If you need any help, please email dev@ignite.apache.org or ask anу advice on http://asf.slack.com _#ignite_ channel.

### CODE REVIEW

File: modules/core/src/test/java/org/apache/ignite/internal/RemoveEntryProcessorTransactionTest.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+        var c = startGrid(0).createCache(new CacheConfiguration<String, Integer>()
+            .setName(DEFAULT_CACHE_NAME)
+            .setAtomicityMode(CacheAtomicityMode.TRANSACTIONAL)
+        );
+
+        for (TransactionConcurrency txConcurrency : TransactionConcurrency.values()) {
+            for (TransactionIsolation txIsolation : TransactionIsolation.values()) {
+                c.put("key", 1);
+
+                try (Transaction tx = grid(0).transactions().txStart(OPTIMISTIC, READ_COMMITTED)) {
```
Comment (alex-plekhanov) [inline modules/core/src/test/java/org/apache/ignite/internal/RemoveEntryProcessorTransactionTest.java:46]: Always OPTIMISTIC, READ_COMMITTED, should be txConcurrency and txIsolation.

File: modules/core/src/main/java/org/apache/ignite/internal/processors/cache/distributed/near/GridNearTxLocal.java [hunk 2]
```diff
@@ -2306,19 +2306,21 @@ private <K, V> Collection<KeyCacheObject> enlistRead(
...
+                                            keepCacheObjects,
+                                            deserializeBinary,
+                                            false,
+                                            getRes,
+                                            readVer,
+                                            0,
+                                            0,
+                                            needVer,
+                                            U.deploymentClassLoader(cctx.kernalContext(), deploymentLdrId));
+                                    }
```
Comment (alex-plekhanov) [inline modules/core/src/main/java/org/apache/ignite/internal/processors/cache/distributed/near/GridNearTxLocal.java:2323]: else missed.put ?

File: modules/core/src/test/java/org/apache/ignite/internal/RemoveEntryProcessorTransactionTest.java (hunk 1 above)
Comment (nizhikov) [reply modules/core/src/test/java/org/apache/ignite/internal/RemoveEntryProcessorTransactionTest.java:46]: Good catch. Thanks.. Fixed.

File: modules/core/src/main/java/org/apache/ignite/internal/processors/cache/distributed/near/GridNearTxLocal.java (hunk 2 above)
Comment (nizhikov) [reply modules/core/src/main/java/org/apache/ignite/internal/processors/cache/distributed/near/GridNearTxLocal.java:2323]: missedKeys means "not found in cache" - so here we in the branch where key was found in cache but removed by entryProcessor.

Looks like no, we don't have to add in missedKey set

**Review by alex-plekhanov — APPROVED** [review]

**Pull request merged**